pypln/backend/workers/spellchecker.py

```
@@ -16,32 +16,43 @@
+    pass
+
+
 class SpellingChecker(PyPLNTask):
     """
     This worker performs spellchecking in the plain text of a document
     """
     def __init__(self):
-        # This method is only called once per process, but that is no problem
-        # since the enchant languange list should not change. Don't use this
-        # method for anything that should run every time the task is called.
-        # See http://docs.celeryproject.org/en/latest/userguide/tasks.html#instantiation
-        # for more information.
-        self.checkers = {lang: SpellChecker(lang) for lang in enchant.list_languages()}
+        # This method is only called once per process
+        self.checkers = {lang: SpellChecker(lang)
+                         for lang in enchant.list_languages()}
 
     def process(self, document):
-        #TODO: this worker may be enhanced by also checking the errors against an specific vocabulary supplied with the document
-        try:
-            checker = self.checkers[document['language']]
-            checker.set_text(document['text'])
-            errors = [[e.word, e.wordpos, e.suggest()] for e in checker]
-        except KeyError:
+        # TODO: this worker may be enhanced by also checking the errors against
+        # an specific vocabulary supplied with the document
+        checker = self.checkers.get(document['language'])
+        if checker is None:
+            # Maybe this should be an exception instead
+            warnings.warn('%s dictionary missing. If running on linux, '
+                          'install the corresponding myspell package'
+                          % document['language'],
+                          MissingDictionaryWarning)
             errors = None
+        else:
+            try:
+                checker.set_text(document['text'])
+                errors = [[e.word, e.wordpos, e.suggest()] for e in checker]
+            except KeyError:
+                errors = None
 
```

I really like the change in the way we treat the case where the specific dictionary does not exist (it wasn't a very good idea to do that catching an exception) and I understand why you kept the try/except block (because there is another possible `KeyError` there). But now that this change was made, I think we can get rid of the entire try/except. This possible `KeyError` is possible in every Task (it represents a pre-requisite not being there when the task runs). I think I prefer to let this exception trickle up so we can see later there was an error (as it happens everywhere else). In short: you found a bug that meant silencing errors that shouldn't be silenced :)